internal/output/disk/disk.go

```diff
@@ -0,0 +1,105 @@
+package disk
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sherifabdlnaby/prism/pkg/types"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+)
+
+//Disk struct
+type Disk struct {
+	FilePath       string
+	FilePermission string
+	TypeCheck      bool
+	Transactions   chan types.Transaction
+	stopChan       chan struct{}
+	logger         zap.Logger
+	wg             sync.WaitGroup
+}
+
+//TransactionChan just return the channel of the transactions
+func (d *Disk) TransactionChan() chan<- types.Transaction {
+	return d.Transactions
+}
+
+//Init func Initialize the disk output plugin
+func (d *Disk) Init(config types.Config, logger zap.Logger) error {
+	if d.FilePath, d.TypeCheck = config["filepath"].(string); !d.TypeCheck {
+		return errors.New("FilePath must be a string")
+	}
+	path.Clean(d.FilePath)
+	if d.FilePermission, d.TypeCheck = config["permission"].(string); !d.TypeCheck {
+		return errors.New("FilePermission must be from a string")
+	}
+	d.Transactions = make(chan types.Transaction)
+	d.stopChan = make(chan struct{})
+	d.logger = logger
+	return nil
+}
+
+//WriteOnDisk func takes the transaction
+//that to be written on the disk
+func (d *Disk) WriteOnDisk(transaction types.Transaction) {
+	defer d.wg.Done()
+	ack := true
+	bytes, err := ioutil.ReadAll(transaction)
+	if err == nil {
+		dir := filepath.Dir(d.FilePath)
+		if _, err = os.Stat(dir); os.IsNotExist(err) {
+			err = os.MkdirAll(dir, os.ModePerm)
+		}
+	}
+	if err == nil {
+		perm32, errParse := strconv.ParseUint(d.FilePermission, 0, 32)
```

Comment:
This is static, no need to parse it, and cast it at every write, final value better be calculated at `Init()` instead.